packages/angular/src/search-token-server/files/scripts/port-allocator.js

```diff
@@ -0,0 +1,102 @@
+const {resolve} = require('path');
+const {
+  readFileSync,
+  writeFileSync,
+  truncateSync,
+  existsSync,
+  appendFileSync,
+} = require('fs');
+const {EOL} = require('os');
+const getPort = require('get-port');
+const {config} = require('dotenv');
+const pathToServerEnv = resolve('server', '.env');
+const environment = config({path: pathToServerEnv}).parsed;
+
+const preferedWebAppPort = 4200;
+const portRangeFallback = [4000, 4999];
+
+const ensureEnvironmentFile = () => {
+  if (!existsSync(pathToServerEnv)) {
```

Comment:
Let's stop the process immediately if the `.env` file is missing instead of running an invalid project.
It also helps with the E2E tests.